llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -5140,14 +5140,11 @@ bool SelectionDAG::isGuaranteedNotToBeUndefOrPoison(SDValue Op, bool PoisonOnly,
   if (Op.getOpcode() == ISD::FREEZE)
     return true;
 
-  // TODO: Assume we don't know anything for now.
   EVT VT = Op.getValueType();
-  if (VT.isScalableVector())
-    return false;
-
-  APInt DemandedElts = VT.isVector()
-                           ? APInt::getAllOnes(VT.getVectorNumElements())
-                           : APInt(1, 1);
+  APInt DemandedElts =
+      VT.isVector() ? APInt::getAllOnes(
```

Comment:
SimplifyDemandedBits does the same thing but writes it as

```
  APInt DemandedElts = VT.isFixedLengthVector()                                  
                           ? APInt::getAllOnes(VT.getVectorNumElements())        
                           : APInt(1, 1);
```

Reply:
That's nicer, thanks, I'll copy that version instead.